File: Exercism/java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

	boolean isArmstrongNumber(int numberToCheck) {
		int n = numberToCheck;
		int countNumbers = 0;
		int sum = 0;
		while (n != 0) {
			n /= 10;
			countNumbers++;
		}
		n = numberToCheck;
		for (int digit = 0; n != 0 ; n /= 10) {
			digit = n % 10;
			sum += Math.pow(digit, countNumbers);
		}
		return numberToCheck == sum;
	}

}

File: Exercism/c/armstrong-numbers/src/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate) {
    int num = candidate;
    int sum = 0;
    int digits = number_of_digits(num);
    while (num != 0) {
        sum += pow(num % 10, digits);
        num /= 10;
    }
    return sum == candidate;
}

int number_of_digits(int num) {
    int n = num;
    int amount = 0;
    while (n != 0) {
        amount++;
        n /= 10;
    }
    return amount;
}
